providers/reddit/reddit_test.go

```diff
@@ -0,0 +1,28 @@
+package reddit
+
+import (
+	"fmt"
+	"testing"
+)
+
+// TestBuildURI makes sure the BuildURI method builds valid URIs.
+func TestBuildURI(t *testing.T) {
+	p := Provider{}
+	cases := []struct {
+		query, want string
+	}{
+		// Regular reddit search.
+		{"best startups",
+			fmt.Sprintf("https://www.reddit.com/search?q=%s", "best+startups")},
+		// Subreddit search.
+		{"/r/cscareerquestions best startups",
```

Comment:
Yeah so I was thinking about this case a bit. If they only put the subreddit we have 2 options.

1. Error out in some way and say the provider didn't return a uri.
2. Search reddits main search for the subreddit name, as they didn't provide a query.

I think I am leaning toward #2.

That way if I do

```
s -p reddit /r/ruby
```

it will show me all subreddits that start with ruby =)